arbor/mc_cell_group.cpp

```diff
@@ -241,6 +241,51 @@ void run_samples(
     sc.sampler({sc.probe_id, sc.tag, sc.index, p.get_metadata_ptr()}, n_sample, sample_records.data());
 }
 
+void run_samples(
+    const fvm_probe_interpolated_multi& p,
+    const sampler_call_info& sc,
+    const fvm_value_type* raw_times,
+    const fvm_value_type* raw_samples,
+    std::vector<sample_record>& sample_records,
+    fvm_probe_scratch& scratch)
+{
+    const sample_size_type n_raw_per_sample = p.raw_handles.size();
+    const sample_size_type n_interp_per_sample = n_raw_per_sample/2;
+    sample_size_type n_sample = (sc.end_offset-sc.begin_offset)/n_raw_per_sample;
+    arb_assert((sc.end_offset-sc.begin_offset)==n_sample*n_raw_per_sample);
+    arb_assert((unsigned)n_interp_per_sample==p.coef[0].size());
+    arb_assert((unsigned)n_interp_per_sample==p.coef[1].size());
+
+    auto& sample_ranges = std::get<std::vector<cable_sample_range>>(scratch);
+    sample_ranges.clear();
+    sample_records.clear();
+
+    auto& tmp = std::get<std::vector<double>>(scratch);
+    tmp.clear();
+    tmp.reserve(n_interp_per_sample*n_sample);
+
+    for (sample_size_type j = 0; j<n_sample; ++j) {
+        auto offset = j*n_raw_per_sample+sc.begin_offset;
+        for (sample_size_type i = 0; i<n_interp_per_sample; ++i) {
+            tmp.push_back(raw_samples[offset+i]*p.coef[0][i]+raw_samples[offset+n_interp_per_sample+i]*p.coef[1][i]);
```

Comment:
Maybe untangle this a bit by putting `offset + i` into variable, it's quite dense.